sdks/python/apache_beam/examples/snippets/transforms/element_wise/regex.py

```diff
@@ -0,0 +1,161 @@
+# coding=utf-8
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+
+def regex_matches(test=None):
+  # [START regex_matches]
+  import apache_beam as beam
+  import re
+
+  def parse_plant(text):
+    m = re.match(r'^([^\s-]+)\s*-\s*(\w+)\s*-\s*(?P<duration>\w+)$', text)
+    if m:
+      yield {
+          'match': m.group(0),              # contains the entire matched text
+          'icon': m.group(1),               # ([^\s-]+) - group
+          'name': m.group(2),               # (\w+) - group
+          'duration': m.group('duration'),  # (?P<duration>\w+) - named group
+      }
+
+  with beam.Pipeline() as pipeline:
+    plant_matches = (
+        pipeline
+        | 'Garden plants' >> beam.Create([
+            '🍓   -   Strawberry   -   perennial',
+            '🥕 - Carrot - biennial',
+            '# 🍌 - invalid - format',
+            '🍆\t-\tEggplant\t-\tperennial',
+            '🍅 - Tomato - annual',
+            '🍉 - invalid - format with trailing words',
+            '🥔-Potato-perennial',
+        ])
+        | 'Parse plants' >> beam.FlatMap(parse_plant)
+        | beam.Map(print)
```

Comment:
Why do we need this Map transform? What's the expected input/output of it?

Reply:
This is a sample, when people copy-paste this and run it they should see the results in their machines. It's using the `print` function which prints to `stdout` and returns `None`. This is just an easier way to demonstrate the transforms without actually going to the filesystem and explaining entire (potentially new) concepts.

It only works on the DirectRunner, but we will eventually generate Jupyter Notebooks from these files too. Since the notebooks use the DirectRunner, it's actually a great experience to interactively see the results with a simple print function.

For testing, the `print` function is replaced by `lambda elem: elem` which makes it a NoOp and just returns the same element so we can then test the resulting PCollection